src/extensions/surveys.tsx

```diff
@@ -304,86 +285,58 @@ export function Questions({
         survey.appearance?.backgroundColor || defaultSurveyAppearance.backgroundColor
     )
     const [questionsResponses, setQuestionsResponses] = useState({})
-    const { readOnly, previewQuestionIndex } = useContext(SurveyContext)
-    const [currentQuestion, setCurrentQuestion] = useState(readOnly ? previewQuestionIndex : 0)
+    const { previewPageIndex } = useContext(SurveyContext)
+    const [currentQuestionIndex, setCurrentQuestionIndex] = useState(previewPageIndex || 0)
     const surveyQuestions = useMemo(() => getDisplayOrderQuestions(survey), [survey])
 
-    const onNextClick = (res: string | string[] | number | null, idx: number) => {
-        const responseKey = idx === 0 ? `$survey_response` : `$survey_response_${idx}`
-        if (idx === survey.questions.length - 1) {
+    // Sync preview state
+    useEffect(() => {
+        setCurrentQuestionIndex(previewPageIndex ?? 0)
+    }, [previewPageIndex])
+
+    const onNextButtonClick = (res: string | string[] | number | null, questionIndex: number) => {
+        const isFirstQuestion = questionIndex === 0
+        const isLastQuestion = questionIndex === survey.questions.length - 1
```

Comment:
~~I think this should be `currentQuestionIndex`, will break otherwise as with shuffled questions, the last questionIndex can show up earlier.~~

This is very confusing to me, because we're calling three different indexes the same thing:

1. The originalQuestionIndex - this is used to route the right response to the right `$survey_response_{originalQuestionIndex}` property, so we're tracking correctly. We need this when shuffling questions.
2. The currentQuestionIndex - this is the index of the question we're currently displaying, *after shuffling the questions*.
3. nextQuestionIndexToShow - the questionIndex inside `onNextButtonClick` which we need to show next. Also relates to index of shuffled questions, not the original question index.

_Maybe_ this works perfectly as is, but I'm having a hard time parsing